### DISCUSSION TITLE
[ZK Stack] - Allow an existing running chain to change DA

### DISCUSSION THREAD

**Discussion by thomas-nguy** [zkSync-Community-Hub/zksync-developers#755] [Feature requests]
### Team or Project

Cronos

### Network

ZK Stack

### Provide a brief description of the feature you would like to see implemented.

The current code prevents you to change the PubdataPricingMode (Rollup or Validium) once the chain has been initialized.
I saw that in recents commit, the limitation has been lifted.

It would be nice to be able to change the mode (validium to rollup) or (rollup to validium) after initialization.

Why?

A lot of chains are choosing to run as a Validium with the assumption that it will be cheaper (or will be cheaper) than running as a Rollup. It has been proven wrong today as we can see the blob fee is nearly zero.
There is also no guarantee that Rollup will ever be more expensive than a Validium in the near future.

The cost of running the proof of inclusion on L1 is a huge downside that may never be justified unless blob fee start to reach some threshold. 

With such uncertainty, chain may want to post the DA on Ethereum when blob are cheap (ie shift to become a Rollup)

Additionally posting on Ethereum should always be considered as an alternative option in case the DA layer is faulty and there should be no reason to prevent the chain to do it. 

Consideration: 

- Security: Switching DA may be only allowed to the STM so that chain operators can't abuse this feature and break security assumptions on their chain.

- Implementation: May not be trivial to switch mode without a breaking changes. 

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

Protocol

**Comment by bxpana** [comment]
It's like you're reading minds. We just updated the docs for ZK Stack regarding validiums and added an [FAQ](https://docs.zksync.io/zk-stack/running-a-zk-chain/validium#faq) about switching DA Layers

> Yes, it is possible.
> - For stage 0 and stage 1 Validiums it is enough to change the client configuration and secrets.
> - For stage 2 Validiums, the operator would need to deploy new L1/L2DAValidator contracts and change their addresses in the contracts that call them.

> The exact details will be provided later, when the process is fully established.

So we'll have more details about that when it's available.

**Comment by thomas-nguy** [comment]
Thanks for the reply @bxpana 

Can you clarify the definition of Stage X Validiums? AFAIK there is no stage definition for Validium (at least this what l2beat claim). 

In case of current state of zkstack, I assume that all ZK-chain are either stage 0 or stage1.
Do that mean that for already running ZK-chain on testnet/mainnet, a simple change in configuration and secrets suffice?

**Reply by thomas-nguy** [reply]
Nvm for the definition, it is explained in this page https://docs.zksync.io/zk-stack/running-a-zk-chain/validium#faq

Thanks

**Reply by bxpana** [reply]
Our pleasure. Happy building!
